api/v1/client.go

```diff
@@ -7,83 +7,96 @@ import (
 	"strconv"
 	"time"
 
+	"github.com/dgraph-io/ristretto"
 	"github.com/go-chi/chi/v5"
-	"github.com/iancoleman/strcase"
-	oasisErrors "github.com/oasisprotocol/oasis-core/go/common/errors"
 
-	beacon "github.com/oasisprotocol/oasis-core/go/beacon/api"
+	governance "github.com/oasisprotocol/oasis-core/go/governance/api"
 	staking "github.com/oasisprotocol/oasis-core/go/staking/api"
-
-	"github.com/oasisprotocol/oasis-indexer/analyzer/util"
 	"github.com/oasisprotocol/oasis-indexer/api/common"
 	"github.com/oasisprotocol/oasis-indexer/log"
-	"github.com/oasisprotocol/oasis-indexer/storage"
+	storage "github.com/oasisprotocol/oasis-indexer/storage/client"
 )
 
 const (
-	tpsWindowSizeMinutes = 5
+	blockCost = 1
+	txCost    = 1
 )
 
 // storageClient is a wrapper around a storage.TargetStorage
 // with knowledge of network semantics.
 type storageClient struct {
-	db     storage.TargetStorage
+	storage *storage.StorageClient
+
+	blockCache *ristretto.Cache
+	txCache    *ristretto.Cache
+
 	logger *log.Logger
 }
 
 // newStorageClient creates a new storage client.
-func newStorageClient(db storage.TargetStorage, l *log.Logger) *storageClient {
-	return &storageClient{db, l}
+func newStorageClient(s *storage.StorageClient, l *log.Logger) (*storageClient, error) {
+	blockCache, err := ristretto.NewCache(&ristretto.Config{
+		NumCounters:        1024 * 10,
+		MaxCost:            1024,
```

Comment:
If I'm understanding this correctly, the `* 10` is because [ristretto suggests 10x your expected cache size](https://github.com/dgraph-io/ristretto#config) for best performance. If that's the case, we should get feedback for how many blocks/transactions worth of data is worthwhile to cache from explorer/wallets. 

Referencing oasisscan as a standard explorer experience, they seem to display the 10 most recent blocks/transactions on their [homepage](https://www.oasisscan.com/), as well as 20 blocks per page on the [blocks page](https://www.oasisscan.com/blocks).

All this to say that I think tbh 1024 is completely acceptable but also that we can probably get away with smaller, should we want it :P

Reply:
Sounds good, I'll leave 1024 as a starting value for now and we can tune it once the frontends/wallets start using the indexer